src/server/api/bannerRouter.ts

```diff
@@ -122,8 +122,8 @@ export const buildBannerRouter = (
                           channelFromBannerChannel(test.bannerChannel),
                           productCatalog.get(),
                           promotions.get(),
+                          variant.promoCodes ?? [],
                           variant.choiceCardsSettings ?? undefined,
-                          variant.bannerContent?.cta,
```

Comment:
here, instead of finding a promoCode in the cta url we use the new `promoCodes` field